### PR TITLE
Make more tests rustfixable

### PR DESCRIPTION
**Pull request by Manishearth**
changelog: Fix various lint suggestions

Progress towards https://github.com/rust-lang/rust-clippy/issues/3630

r? @phansch

### CODE REVIEW

File: tests/ui/eq_op.rs [hunk 1]
```diff
@@ -2,7 +2,8 @@
 #[warn(clippy::eq_op)]
 #[allow(clippy::identity_op, clippy::double_parens, clippy::many_single_char_names)]
 #[allow(clippy::no_effect, unused_variables, clippy::unnecessary_operation, clippy::short_circuit_statement)]
-#[warn(clippy::nonminimal_bool)]
+#[allow(clippy::nonminimal_bool)]
```
Comment (flip1995) [inline tests/ui/eq_op.rs]: I guess this is tested in another file? Everything else LGTM

File: clippy_lints/src/entry.rs [hunk 2]
```diff
@@ -142,13 +145,21 @@ impl<'a, 'tcx, 'b> Visitor<'tcx> for InsertVisitor<'a, 'tcx, 'b> {
...
+                let maybe_semi = if self.has_else {
+                    ""
+                } else {
+                    ";"
+                };
                 span_lint_and_then(self.cx, MAP_ENTRY, self.span,
                                    &format!("usage of `contains_key` followed by `insert` on a `{}`", self.ty), |db| {
                     if self.sole_expr {
-                        let help = format!("{}.entry({}).or_insert({})",
+                        let help = format!("{}.entry({}).or_insert({}){}",
```
Comment (phansch) [inline clippy_lints/src/entry.rs]: Dogfood says that these two if branches are now identical. Is that what we want? What was the reason for not including `or_insert({})` in the `else` branch of the original code?

File: tests/ui/for_loop_unfixable.stderr [hunk 3]
```diff
@@ -0,0 +1,34 @@
+error[E0423]: expected value, found macro `vec`
+  --> $DIR/for_loop_unfixable.rs:27:15
+   |
+LL |     for _v in vec.iter().next() {}
+   |               ^^^ help: use `!` to invoke the macro: `vec!`
```
Comment (phansch) [inline tests/ui/for_loop_unfixable.stderr]: The test file is missing the `vec` local.

File: tests/ui/eq_op.rs (hunk 1 above)
Comment (Manishearth) [reply tests/ui/eq_op.rs]: Yep

File: clippy_lints/src/entry.rs (hunk 2 above)
Comment (Manishearth) [reply clippy_lints/src/entry.rs]: Ah, I should be splitting this file, I made a mistake

Comment (Manishearth) [reply clippy_lints/src/entry.rs]: Removed the commit, will deal with later.

Comment (Manishearth) [reply clippy_lints/src/entry.rs]: We need multiple suggestions for the other one, it's going to be tricky

**Comment by flip1995** [comment]
`./util/dev fmt`

**Comment by Manishearth** [comment]
This is what i get for making a PR in a hurry because I didn't think I'd be back on my laptop until after my flight

**Pull request merged**